tests/common/helpers/snmp_helpers.py

```diff
@@ -10,20 +10,34 @@
 
 DEF_WAIT_TIMEOUT = 300
 DEF_CHECK_INTERVAL = 10
+SNMP_SUBAGENT_WAIT_TIMEOUT = 120
+SNMP_SUBAGENT_CHECK_INTERVAL = 5
 
 global_snmp_facts = {}
 
 
+def is_snmp_subagent_running(duthost):
+    cmd = "docker exec snmp supervisorctl status snmp-subagent"
+    output = duthost.shell(cmd)
+    if "RUNNING" in output["stdout"]:
+        return True
+    return False
+
+
 def _get_snmp_facts(localhost, host, version, community, is_dell, include_swap, module_ignore_errors):
     snmp_facts = localhost.snmp_facts(host=host, version=version, community=community, is_dell=is_dell,
                                       module_ignore_errors=module_ignore_errors, include_swap=include_swap)
     return snmp_facts
 
 
-def _update_snmp_facts(localhost, host, version, community, is_dell, include_swap):
+def _update_snmp_facts(localhost, host, version, community, is_dell, include_swap, duthost):
     global global_snmp_facts
 
     try:
+        pytest_assert(
+            wait_until(SNMP_SUBAGENT_WAIT_TIMEOUT, SNMP_SUBAGENT_CHECK_INTERVAL, 0,
+                       is_snmp_subagent_running, duthost),
+            "SNMP Sub-Agent is not in Running state")
```

Comment:
Thanks for the fix, please also attach the test result, thanks!

Reply:
Summary of Tests Run on our Testbed ->

```
=========================== short test summary info ============================
SKIPPED [1] snmp/test_snmp_fdb.py:95: Not supporteds on SONiC leaf-fanout platform
SKIPPED [1] snmp/test_snmp_memory.py:174: Swap is not on for this device, snmp does not support swap related queries when swap isn't on
SKIPPED [1] snmp/test_snmp_phy_entity.py:367: Not supported on non supervisor node
SKIPPED [1] platform_tests/thermal_control_test_helper.py:123: No mocker defined for this platform %s
SKIPPED [1] snmp/test_snmp_phy_entity.py:636: psu_controller is None, skipping this test
FAILED snmp/test_snmp_psu.py::test_snmp_numpsu[mth-t0-64]
FAILED snmp/test_snmp_queue_counters.py::test_snmp_queue_counters[mth-t0-64]
======= 2 failed, 19 passed, 5 skipped, 1 warning in 2183.04s (0:36:23) ========
```

Reply:
Thanks @agadia-cisco ,
Next time please paste the test result in the PR description,
And for the 2 failures in the output,
I believe there were here and not related to your change, right?